challenges.js

```diff
@@ -115,3 +115,45 @@ const challengesContainer = document.querySelector(".challenges-container");
 
 let view = new ChallengeListView();
 view.render(challengesContainer);
+
+//function keyword filter
+
+function keyFilter() {
+  // Declare variables
+  var input, filter, challenges, title, infoText, i;
```

Comment:
Using `var` is very uncommon nowadays. Before 2015 (when `const` and `let` were introduced) it was the only way to declare a variable. It took some time for the JS community to transition, but now everyone uses `const` and `let` unless they really need `var`.

The main differences between the keywords are illustrated in this table:

|Trait|`var`|`let`|`const`|
|-|-|-|-|
|Supports reassignment|yes|yes|no|
|Scope|function|block|block|

You can read about [scoping on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const#block_scoping).

The other thing that you're doing here that is a bit uncommon today is to declare all the variables upfront. I rarely see this. Most people declare the variables the first time they use them (unless there is some special need for declaring them upfront). There's both an element of personal preference, and some objective reasons why declaring variables as you use them is better.